File: lib/contant/log_level.dart
import 'dart:ui';

class LogLevel {
  static const int panic = 0;
  static const int fatal = 1;
  static const int error = 2;

  static const int warn = 3;
  static const int info = 4;
  static const int debug = 5;
  static const int trace = 6;

  static Color toColor(int level) {
    //Color.fromARGB(a, r, g, b)
    return switch(level) {
      LogLevel.panic => const Color.fromARGB(255, 255, 0, 0),
      LogLevel.fatal => const Color.fromARGB(255, 255, 0, 0),
      LogLevel.error => const Color.fromARGB(255, 255, 0, 0),
      LogLevel.warn => const Color.fromARGB(255, 255, 165, 0),
      LogLevel.info => const Color.fromARGB(255, 0, 0, 255),
      LogLevel.debug => const Color.fromARGB(255, 0, 255, 0),
      LogLevel.trace => const Color.fromARGB(255, 0, 255, 0),
      _ => const Color.fromARGB(255, 0, 0, 0)
    };
  }

  static String toStr(int level) {
    return switch(level) {
      LogLevel.panic => "Panic",
      LogLevel.fatal => "Fatal",
      LogLevel.error => "Error",
      LogLevel.warn => "Warn",
      LogLevel.info => "Info",
      LogLevel.debug => "Debug",
      LogLevel.trace => "Trace",
      _ => ""
    };
  }
}

File: android/app/src/main/kotlin/com/github/jing332/alistflutter/constant/LogLevel.kt
package com.github.jing332.alistflutter.constant

import androidx.annotation.IntDef

@IntDef(
    LogLevel.PANIC,
    LogLevel.FATAL,
    LogLevel.ERROR,
    LogLevel.WARN,
    LogLevel.INFO,
    LogLevel.DEBUG,
    LogLevel.TRACE
)
annotation class LogLevel {
    companion object {
        const val PANIC = 0
        const val FATAL = 1
        const val ERROR = 2
        const val WARN = 3
        const val INFO = 4
        const val DEBUG = 5
        const val TRACE = 6

        fun Int.toLevelString(): String {
            return when (this) {
                PANIC -> "PANIC"
                FATAL -> "FATAL"
                ERROR -> "ERROR"
                WARN -> "WARN"
                INFO -> "INFO"
                DEBUG -> "DEBUG"
                TRACE -> "TRACE"
                else -> "UNKNOWN"
            }
        }
    }

}